### PR TITLE
refactor: Add dynamic plugin loading for enterprise components

### PR DESCRIPTION
**Pull request by hari-kuriakose**
## What

- Add dynamic plugin loading support to OSS codebase
- Enable enterprise components to be loaded at runtime without modifying tracked files
- Add explicit Traefik port labels for proper service routing
- Clean up run-platform.sh service ignore list

## Why

- Enterprise code was overwriting git-tracked OSS files causing dirty git state
- Need clean separation between OSS and enterprise codebases
- OSS should work independently without enterprise components
- Traefik needs explicit port configuration for reliable container routing
- Service ignore list needed updating with additional spawned services

## How

- `unstract_migrations.py`: Uses try/except ImportError to load from `pluggable_apps.migrations_ext`
- `api_hub_usage_utils.py`: Uses try/except ImportError to load from `plugins.verticals_usage`
- `utils.py`: Uses try/except ImportError to load from `pluggable_apps.manual_review_v2` and `plugins.workflow_manager.workflow_v2.rule_engine`
- `backend.Dockerfile`: Conditional install of `requirements.txt` if present
- `docker-compose.yaml`: Add `traefik.http.services.*.loadbalancer.server.port` labels for backend (8000) and frontend (3000)
- `run-platform.sh`: Rename `spawned_services` to `ignored_services` and extend list with tool-classifier, tool-text_extractor, worker-unified

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why.

- **Yes, this PR carries regression risk.** While the changes are designed to be backward-compatible, refactoring core import mechanisms and modifying how plugins are loaded can introduce subtle issues:
  - Import order or timing changes could affect module initialization
  - Exception handling in try/except blocks may mask unexpected errors
  - Default fallback behavior may differ slightly from original implementations
  - Changes to Traefik port labels could affect routing if misconfigured

## Reviewer Guide: Analyzing Refactoring Impact

When reviewing this PR, please pay special attention to the following areas:

1. **Import path changes**: Verify that the try/except ImportError patterns correctly fall back to the original behavior. Test both with and without enterprise plugins present.

2. **Migration loading** (`unstract_migrations.py`): Confirm that database migrations run correctly in both OSS and enterprise configurations. Migration failures can be difficult to recover from.

3. **Usage utilities** (`api_hub_usage_utils.py`, `utils.py`): Check that the fallback implementations match the expected interfaces. Any mismatch could cause runtime errors in production.

4. **Dockerfile changes**: Verify that the conditional `requirements.txt` install doesn't break builds when the file is absent or empty.

5. **Traefik configuration**: Test that services are accessible via the expected routes. Consider edge cases like container restarts and service discovery timing.

6. **Service ignore list**: Confirm that the renamed variable and extended list don't inadvertently exclude or include wrong services.

## Database Migrations

- None

## Env Config

- None

## Relevant Docs

- None

## Related Issues or PRs

- None

## Dependencies Versions

- None

## Notes on Testing

- OSS build: Verify app starts and works without enterprise plugins
- Enterprise build: Verify plugins are loaded and function correctly
- Docker: Verify Traefik routes traffic correctly to backend and frontend services

## Screenshots

- N/A (backend/infrastructure changes only)

## Checklist

I have read and understood the [Contribution Guidelines](https://docs.unstract.com/unstract/contributing/unstract/).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/plugins/workflow_manager/workflow_v2/utils.py [hunk 1]
```diff
@@ -1,64 +1,97 @@
...
-        pass
+        try:
+            import random
+
+            from pluggable_apps.manual_review_v2.helper import get_db_rules_by_workflow_id
+        except ImportError:
+            return None
+
+        num_to_select = max(1, int(n * (percentage / 100)))
+        return set(random.sample(range(1, n + 1), num_to_select))
```
[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply backend/plugins/workflow_manager/workflow_v2/utils.py]: Fixed

File: backend/plugins/workflow_manager/workflow_v2/api_hub_usage_utils.py [hunk 2]
```diff
@@ -1,61 +1,79 @@
...
 
         Returns:
-            bool: False in OSS version (usage tracking not available).
-                  Enterprise version returns True if tracking succeeded.
+            bool: True if tracking succeeded, False otherwise.
         """
-        # OSS version - no usage tracking available
-        logger.debug("API hub usage tracking not available in OSS version")
-        return False
+        verticals_usage_plugin = get_plugin("verticals_usage")
```
Comment (chandrasekharan-zipstack) [inline backend/plugins/workflow_manager/workflow_v2/api_hub_usage_utils.py:33]: NIT: Related to [UN-3147](https://zipstack.atlassian.net/browse/UN-3147?atlOrigin=eyJpIjoiZmYyYjFmMDk5MTI0NDU4NDhhZjQzNGVkOTRlMGJmMjYiLCJwIjoiamlyYS1zbGFjay1pbnQifQ) and can be done in the future but we could probably move out this check at the place where this plugin is called to make the code more readable

**Review by jaseemjaskp — APPROVED** [review]

File: backend/plugins/workflow_manager/workflow_v2/api_hub_usage_utils.py [hunk 3]
```diff
@@ -65,16 +83,21 @@ def cache_api_hub_headers(
...
+        if not verticals_usage_plugin:
+            return False
+
+        try:
+            headers_cache = verticals_usage_plugin["headers_cache_class"]()
+            return headers_cache.store_headers(execution_id, headers)
+        except Exception as e:
+            logger.error(f"Error caching API hub headers: {e}")
+            return False
```
[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply backend/plugins/workflow_manager/workflow_v2/api_hub_usage_utils.py:103]: @jaseemjaskp Want to keep the current logic intact for now, please check this later.

File: backend/plugins/workflow_manager/workflow_v2/utils.py [hunk 4]
```diff
@@ -1,64 +1,91 @@
...
             percentage (float): The percentage of files to select.
             n (int): The total number of files.
 
         Returns:
-            Any: The method is currently a placeholder and does not return a value.
+            A set of file indices, or None if manual_review_v2 is not available.
         """
-        pass
+        num_to_select = max(1, int(n * (percentage / 100)))
+        return set(random.sample(range(1, n + 1), num_to_select))
```
[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply backend/plugins/workflow_manager/workflow_v2/utils.py:40]: @muhammad-ali-e Want to keep the current logic intact for now, please check this later.

File: backend/plugins/workflow_manager/workflow_v2/utils.py [hunk 5]
```diff
@@ -1,64 +1,91 @@
...
+            return file_hash
+
+        try:
+            from workflow_manager.endpoint_v2.models import WorkflowEndpoint
+        except ImportError:
+            return file_hash
+
+        if index in q_file_no_list:
+            file_hash.file_destination = WorkflowEndpoint.ConnectionType.MANUALREVIEW
         return file_hash
```
[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply backend/plugins/workflow_manager/workflow_v2/utils.py:89]: @muhammad-ali-e Want to keep the current logic intact for now, please check this later.

File: backend/plugins/workflow_manager/workflow_v2/api_hub_usage_utils.py (hunk 2 above)
Comment (hari-kuriakose) [reply backend/plugins/workflow_manager/workflow_v2/api_hub_usage_utils.py:33]: @jaseemjaskp Want to keep the current logic intact for now, please check this later.
